src/cookieconsent.js

```diff
@@ -1005,60 +1037,78 @@
     // When the service uses a script, the chances are that you'll have to use the script to make additional requests. In these
     // cases, the services `callback` property is called with a `done` function. When performing async operations, this must be called
     // with the data (or Error), and `cookieconsent.locate` will take care of the rest
-    var locationServices = {
-      ipinfo: function (obj) {
-        return {
-          // This service responds with JSON, so we simply need to parse it and return the country code
-          url: 'http://ipinfo.io',
-          headers: ['Accept: application/json'],
-          callback: function (done, response) {
-            var json = JSON.parse(response);
-            return json.error ? toError(json) : {
-              code: json.country
-            };
-          }
-        }
-      },
-      freegeoip: function () {
-        return {
-          // This service responds with JSON, but they do not have CORS set, so we must use JSONP and provide a callback
-          // The callback MUST BE `cookieconsent.locate.jsonp(SERVICE_NAME)` (so cookieconsent.locate.jsonp('freegeoip') for this index)
-          url: '//freegeoip.net/json/?callback={callback}',
-          isScript: true, // this is JSONP, therefore we must set it to run as a script
-          callback: function (done, response) {
-            var json = JSON.parse(response);
-            return json.error ? toError(json) : {
-              code: json.country_code
-            };
+    var defaultOptions = {
+      // the order that services will be attempted in
+      services: [
+        'freegeoip',
+        'ipinfo',
+        'maxmind',
+
+        /* OPTIONAL
+
+        function () {
+          return serviceDefinition
+        },
+
+        {name: 'myService', ...serviceDefinition},
+
+        */
+      ],
+      serviceDefinitions: {
+        ipinfo: function() {
```

Comment:
the idea is that the user passes it as an option, so we should be able to do something like:

```
ipinfo: function (options) {
  if (options.API_KEY) ...
}
```